tests/model_serving/maas_billing/multitenancy/aigateway/utils.py

```
@@ -387,45 +460,116 @@ def _wait_until_resource_absent(
+) -> None:
+    """Assert tenant-admin and object-admin Roles and RoleBindings were removed after AITenant deletion."""
+    tenant_admin_name = tenant_admin_role_name(aigateway_name=aigateway_name)
+    object_admin_name = aigateway_object_admin_role_name(aigateway_name=aigateway_name)
+    for namespace, resource_name, resource_kind in (
+        (tenant_namespace_name, tenant_admin_name, "Role"),
+        (tenant_namespace_name, tenant_admin_name, "RoleBinding"),
+        (infra_namespace, object_admin_name, "Role"),
+        (infra_namespace, object_admin_name, "RoleBinding"),
+    ):
+        if resource_kind == "Role":
+            exists_check = lambda bound_namespace=namespace, role_name=resource_name: Role(
```

Flake8 (7.3.0)

[error] do not assign a lambda expression, use a def

(E731)